core/src/main/java/com/novoda/merlin/receiver/ConnectivityCallbacks.java

```diff
@@ -3,20 +3,21 @@
 import android.annotation.TargetApi;
 import android.net.ConnectivityManager;
 import android.net.Network;
-import android.net.NetworkInfo;
 import android.os.Build;
 
+import com.novoda.merlin.service.ConnectivityChangeEventExtractor;
 import com.novoda.merlin.service.MerlinService;
 
 @TargetApi(Build.VERSION_CODES.LOLLIPOP)
 class ConnectivityCallbacks extends ConnectivityManager.NetworkCallback {
 
-    private final ConnectivityManager connectivityManager;
-    private final MerlinService.ConnectivityChangesListener connectivityChangesListener;
+    private final MerlinService.ConnectivityChangesNotifier connectivityChangesNotifier;
+    private final ConnectivityChangeEventExtractor connectivityChangeEventExtractor;
```

Comment:
`Collaborator` for extracting `NetworkInfo` and creating a `ConnectivityChangeEvent` from `Network`.